connector/kinesis-asl-assembly/pom.xml

```diff
@@ -62,12 +62,18 @@
     <dependency>
       <groupId>com.google.protobuf</groupId>
       <artifactId>protobuf-java</artifactId>
-      <version>2.6.1</version>
-      <!-- 
-         We are being explicit about version here and overriding the 
-         spark default of 2.5.0 because KCL appears to have introduced 
-         a dependency on protobuf 2.6.1 somewhere between KCL 1.4.0 and 1.6.1.
-       -->
+      <scope>compile</scope>
```

Comment:
I'm glad to see that Kinesis can use the same version of protobuf-java as other modules, but is this feasible?

Reply:
Hi @LuciferYang, I saw similar patterns in other modules as well: https://github.com/apache/spark/blob/v3.5.0/connector/connect/server/pom.xml#L173-L175. I believe this should work as long as we shade it.

Reply:
For the record, I removed this shaded `protobuf-java` completely in Apache Spark 4.0.0 independently.
-  #45096

Reply:
Thanks @dongjoon-hyun for letting me know. Updated the PR to use the same protobuf-java version as specified in `<protobuf.version>`.  I had to change the scope to "compile" to fix ClassNotFoundExceptions.